src/main/java/com/blakebr0/extendedcrafting/compat/jei/endercrafter/EnderCrafterCategory.java

```diff
@@ -77,13 +78,25 @@ public void setRecipe(IRecipeLayout layout, IRecipeWrapper wrapper, IIngredients
 			}
 		}
 
-		int xd = 1;
-		for (List<ItemStack> stack : inputs) {
-			stacks.set(xd, stack);
-			xd++;
-		}
-		
-		if (wrapper instanceof TableShapelessWrapper) {
+		if (wrapper instanceof TableShapedWrapper) {
+			TableShapedWrapper shaped = (TableShapedWrapper) wrapper;
+
+			int stackIndex = 0;
+			for (int i = 0; i < shaped.getHeight(); i++) {
+				for (int j = 0; j < shaped.getWidth(); j++) {
+					int index = 1 + (i * 3) + j;
+
+					stacks.set(index, inputs.get(stackIndex));
+
+					stackIndex++;
+				}
+			}
+		} else if (wrapper instanceof TableShapelessWrapper) {
```

Comment:
I don't believe there is a need for the `else if` here; a recipe is either shaped or shapeless.